### PR TITLE
Rust var as i32

### PR DESCRIPTION
**Pull request by inabe-a**
Adding From<T>, TryFrom<T> impl in Rust for Var for vector types with up to 4 elements.

https://github.com/fragcolor-xyz/shards/issues/263

**Has a test plan**

### CODE REVIEW

**Comment by sinkingsugar** [comment]
`half`/`f16` is interesting but not being "standard" makes it less useful.
Is the rust community mostly converging using `half` crate though? In that case maybe why not.

**Review by guusw — CHANGES_REQUESTED** [review]

**Comment by inabe-a** [comment]
Tested locally, but I haven't enabled `cargo test` here. We probably want to push that separately.
https://discord.com/channels/969754837220737054/969766235967459419/986549166673129512

File: rust/src/types.rs [hunk 1]
```diff
@@ -3280,3 +3290,96 @@ lazy_static! {
...
+
+  test_to_from_vec4!(i16, i16::MAX, "[i16,4] conversion failed");
+  test_to_from_vec4!(i16, i16::MIN, "[i16,4] conversion failed");
+  test_to_from_vec4!(u16, u16::MAX, "[u16,4] conversion failed");
+  test_to_from_vec4!(u16, u16::MIN, "[u16,4] conversion failed");
+  test_to_from_vec4!(half::f16, half::f16::ZERO, "[half::f16,4] conversion failed");
+  test_to_from_vec4!(half::f16, half::f16::MAX, "[half::f16,4] conversion failed");
+  test_to_from_vec4!(half::f16, half::f16::MIN, "[half::f16,4] conversion failed");
+  test_to_from_vec4!(half::f16, half::f16::MIN_POSITIVE, "[half::f16,4] conversion failed");
+  test_to_from_vec4!(half::f16, half::f16::EPSILON, "[half::f16,4] conversion failed");
```
Comment (inabe-a) [inline rust/src/types.rs:3382]: Shards is so sharp, it cuts the epsilon in half 🎉

**Comment by sinkingsugar** [comment]
Considering that I'm now propagating errors et al in #279 we should reconsider the overflow handling. We might wanna error on i64 overflow.

**Pull request merged**